### ISSUE TITLE
OpenCV::imwrite receives 3 params but needs num%2==0

### ISSUE THREAD

**Issue by MaxWinsemius** [szanni/slideextract#6]
When I run the slide extractor, I always receive the following error:

```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.7.0) /usr/src/debug/opencv/opencv-4.7.0/modules/imgcodecs/src/loadsave.cpp:738: error: (-2:Unspecified error) in function 'bool cv::imwrite_(const String&, const std::vector<Mat>&, const std::vector<int>&, bool)'
> Encoding 'params' must be key-value pairs:
>     '(params.size() & 1) == 0'
> where
>     'params.size()' is 3

zsh: IOT instruction (core dumped)  slideextract -g cross-validation.mp4 cross-validation
```

It seems [`slideextract.cpp:171`](https://github.com/szanni/slideextract/blob/master/slideextract.cpp#L171) the `imwrite(_, _, params) -> params` argument requires a key-value pairwise array (`[k1, v1, k2, v2, ..., kn, vn]`).

I commented out line [152](https://github.com/szanni/slideextract/blob/master/slideextract.cpp#L152) and then the program runs without errors. I don't know if this is a proper fix or not, since i don't know how OpenCV works.

OpenCV version: 4.7.0

**Comment by thibaultmol** [comment]
Yeah, i'm also getting that

```
slideextract -g "Infosessie： financiën 2022 [M9Vc2bWowgU].webm" outprefix
QSettings::value: Empty key passed
QSettings::value: Empty key passed
Fontconfig warning: ignoring Default.UTF-8: not a valid language tag
Selected ROI: 481.140:1437x804
terminate called after throwing an instance of 'cv::Exception'
what():  OpenCV(4.7.0) /usr/src/debug/opencv/opencv-4.7.0/modules/imgcodecs/src/loadsave.cpp:738: error: (-2:Unspecified error) in function 'bool cv::imwrite_(const String&, const std::vector<Mat>&, const std::vector<int>&, bool)'
> Encoding 'params' must be key-value pairs:
>     '(params.size() & 1) == 0'
> where
>     'params.size()' is 3

fish: Job 1, 'slideextract -g "Infosessie： f…' terminated by signal SIGABRT (Abort)

```

**Comment by szanni** [comment]
Confirmed, thanks for reporting.
OpenCV has been consistently breaking their API and in this case ABI with pretty much every release.

> I commented out line [152](https://github.com/szanni/slideextract/blob/master/slideextract.cpp#L152) and then the program runs without errors. I don't know if this is a proper fix or not, since i don't know how OpenCV works.

It does indeed work on my system and is how the usage is described in the [docs](https://docs.opencv.org/4.7.0/d4/da8/group__imgcodecs.html) .

**Comment by szanni** [comment]
Fixed in `0.3.1` / latest upstream.
